lib/toxiproxy_ex.ex

```diff
@@ -441,11 +405,11 @@ defmodule ToxiproxyEx do
       iex> ToxiproxyEx.version!()
       "2.1.2"
   """
-  @spec version!() :: :ok
+  @spec version!() :: String.t()
   def version!() do
-    case Client.version() do
-      {:ok, %{body: res}} -> res
-      _ -> raise ServerError, message: "Could not fetch version"
+    case Client.request!(:get, "/version") do
+      %{"version" => version} -> version
```

Comment:
New versions of Toxiproxy changed the format of the response to this.

Reply:
Until now I gotten away with not stating compatibility of the underlying Toxiproxy version. Seems like it's time to change that.

Since you have more experience maintaining client libs than me?

Reply:
@Jcambass I would probably make major versions of this library compatible with major versions of Toxiproxy. The `/version` change was a mistake on the part of the Toxiproxy team IMO, since they made a breaking change in a minor release.

Doing this means maintaining a couple of release lines as you said, but I don't think it's much of a burden since Toxiproxy doesn't backport much changes. Security is not a concern in this field IMO, since we're talking mostly about a tool to run local tests.

So, practically:

  1. Release ToxiproxyEx 2.0, which depends on Toxiproxy 2+
  2. Add these little tweaks to make ToxiproxyEx support changes such as the `/version` endpoint in 2.6.0, using `PATCH` instead of `POST` if the version is greater than 2.6.0, and so on.

Thoughts?

Reply:
I like the idea 👍🏻 Thanks for your thoughts!